# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/websocket/PWebSocketConnection.ipdl
# Commit: f7b6aac1b58b
# Full Hash: f7b6aac1b58bc959e431136e1b23231abe55d0b1
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2020-07-16 21:27:37
# Regressor Bug: 1497249
# File Overlap Count: 4
# Description:
#   Bug 1497249 - P2: ipdl for nsIWebSocketConnection r=michal
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D30624
# ==============================================================================

diff -r 74bb3c2ab767 -r f7b6aac1b58b netwerk/protocol/websocket/PWebSocketConnection.ipdl
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/netwerk/protocol/websocket/PWebSocketConnection.ipdl	Thu Jul 16 13:34:58 2020 +0000
@@ -0,0 +1,33 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set sw=2 ts=8 et tw=80 ft=cpp : */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+include protocol PHttpConnectionMgr;
+
+namespace mozilla {
+namespace net {
+
+async refcounted protocol PWebSocketConnection
+{
+  manager PHttpConnectionMgr;
+
+parent:
+  async OnTransportAvailable(nsCString securityInfoSerialization);
+  async OnError(nsresult aStatus);
+  async OnTCPClosed();
+  async OnDataReceived(uint8_t[] aData);
+  async OnUpgradeFailed(nsresult aReason);
+
+child:
+  async EnqueueOutgoingData(uint8_t[] aHeader, uint8_t[] aPayload);
+  async StartReading();
+  async DrainSocketData();
+
+  async __delete__();
+
+};
+
+} //namespace net
+} //namespace mozilla